src/v/cluster/tests/CMakeLists.txt

```diff
@@ -28,7 +28,6 @@ set(srcs
     topic_table_partition_generator_test.cc
     idempotency_tests.cc
     feature_barrier_test.cc
-    tm_stm_tests.cc
```

Comment:
nit on commit message.. we usually try to prefix the summary line with the subsystem: example tests/transaction or something like that (next commit too).